### ISSUE TITLE
Simplify the formulation of SDS loss

### ISSUE THREAD

**Issue by ashawkey** [threestudio-project/threestudio#266]
We can use a even simpler reparameterization to achieve the SDS loss:
```python
# current:
target = (latents - grad).detach()
# d(loss)/d(latents) = latents - target = latents - (latents - grad) = grad
loss_sds = 0.5 * F.mse_loss(latents, target, reduction="sum") / batch_size

# new:
# d(loss)/d(latents) = grad
loss_sds = (latents * grad.detach()).sum()
```
This is originally proposed in https://github.com/ashawkey/stable-dreamfusion/issues/335 (credits to @Xallt).
Maybe we can implement it here too?

**Comment by bennyguo** [comment]
Looks good! Could you verify that the magnitude of the gradient remains the same (especially for `data.batch_size > 1`?)

**Comment by Xallt** [comment]
Yeah this MSE formulation is also cool, and with an even more meaningful loss value. I'd actually prefer this one to the one I suggested

**Comment by Xallt** [comment]
@voletiv Faster in training? Why do you think that happens?
Kernel for the F.mse_loss gradient computation is faster than for the multiplication?

**Comment by voletiv** [comment]
@Xallt By "faster" I don't mean per compute, I mean iterations from initialization to a good final rendering.
Multiple factors contribute to this, possibly unrelated to the loss as such : the combination of implicit-volume and original loss works very well, but SDF and original loss is not as effective (in my experiments for 2D->3D).

**Comment by bennyguo** [comment]
I think @Xallt makes a very good point. Although `latents * grad.detach()` is easier to understand, the MSE formulation gives meaningful gradient magnitude. So I think we'll stick with the MSE formulation then.
